instrumentation/github.com/go-sql-driver/mysql/splunkmysql/test/integration_test.go

```diff
@@ -0,0 +1,219 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package test
+
+// Build restrictions come from docker not being available on Windows and
+// MacOS GitHub Actions. The code itself should be compatible these	systems
+// and the build restrictions can be removed if this is run elsewhere.
+
+import (
+	"context"
+	"database/sql"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/signalfx/splunk-otel-go/instrumentation/database/sql/splunksql"
+	_ "github.com/signalfx/splunk-otel-go/instrumentation/github.com/go-sql-driver/mysql/splunkmysql"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/sdk/trace/tracetest"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+)
+
+const (
+	user     = "testuser"
+	pass     = "testuser-password"
+	rootPass = "secret"
+	host     = "localhost"
+	port     = 3306
+	dbName   = "testdb"
+
+	createStmt = "CREATE TABLE IF NOT EXISTS squareNum ( number integer, squareNumber integer )"
+	queryStmt  = "SELECT squareNumber FROM squareNum WHERE number = ?"
+	insertStmt = "INSERT INTO squareNum VALUES( ?, ? )"
+)
+
+var (
+	dsn          = fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", user, pass, host, port, dbName)
+	dsnRoot      = fmt.Sprintf("root:%s@tcp(%s:%d)/%s", rootPass, host, port, dbName)
+	dsnSanitized = fmt.Sprintf("%s@tcp(%s:%d)/%s", user, host, port, dbName)
+)
+
+func newFixtures(t *testing.T) (*tracetest.SpanRecorder, *trace.TracerProvider, *sql.DB, func(*testing.T)) {
+	sr := tracetest.NewSpanRecorder()
+	tp := trace.NewTracerProvider(trace.WithSpanProcessor(sr))
+	db, err := splunksql.Open("mysql", dsn, splunksql.WithTracerProvider(tp))
+	require.NoError(t, err)
+	return sr, tp, db, func(t *testing.T) {
+		require.NoError(t, db.Close())
+		require.NoError(t, tp.Shutdown(context.Background()))
+	}
```

Comment:
probably we could

```go
	t.Cleanup(func() {
		require.NoError(t, db.Close())
		require.NoError(t, tp.Shutdown(context.Background()))
	})
```

instead of returning the shutdown function